get_coverage_for_challenge.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+set -u
+set -o pipefail
+
+SCRIPT_CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+CHALLENGE_ID=$1
+NODEJS_TEST_REPORT_JSON_FILE="${SCRIPT_CURRENT_DIR}/coverage/coverage-summary.json"
+NODEJS_CODE_COVERAGE_INFO="${SCRIPT_CURRENT_DIR}/coverage.tdl"
+
+( cd ${SCRIPT_CURRENT_DIR} && npm install && npm run coverage || true 1>&2 )
+
+[ -e ${NODEJS_CODE_COVERAGE_INFO} ] && rm ${NODEJS_CODE_COVERAGE_INFO}
+
+if [ -f "${NODEJS_TEST_REPORT_JSON_FILE}" ]; then
+    node_modules/prettier/bin-prettier.js --write ${NODEJS_TEST_REPORT_JSON_FILE}
+
+    COVERAGE_OUTPUT=$(grep ${CHALLENGE_ID} ${NODEJS_TEST_REPORT_JSON_FILE} -A 4 | grep -o -P '(?<="lines": {).*(?=},)' | tr -d ' "')
```

Comment:
This line is failing on my laptop. The grep command is different on Mac:
```
++ grep -o -P '(?<="lines": {).*(?=},)'
++ tr -d ' "'
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
+ COVERAGE_OUTPUT=
```
Let's use JQ for JSON parsing.

Reply:
JQ was my first attempt, but JQ considers the coverage json file as an invalid json file. I;ll try to come up with something works for both.

JQ would have been ideal and more appropriate use in this case.

Reply:
```
jq: error: coverage/0 is not defined at <top-level>, line 1:
coverage/coverage-summary.json
jq: error: coverage/0 is not defined at <top-level>, line 1:
coverage/coverage-summary.json         
jq: error: summary/0 is not defined at <top-level>, line 1:
coverage/coverage-summary.json                  
jq: 3 compile errors
```